### ISSUE TITLE
Invalid request token in forms opened at mobile email app.

### ISSUE THREAD

**Issue by piotrwarsztocki** [contao/contao#7540]
### Affected version(s)

5.2,5.3

### Description

I'm not sure if this is a bug, but I'll describe it just in case.

I've recently had a lot of problems sending a form from one of my sites. I kept getting the **"Invalid request token"** error. After a series of tests, I discovered that the problem is with forms opened on mobile phones directly from a link. The email application opens such links in its own frame by default. A form opened in this way reports an invalid token. When you open the link in your browser application on mobile phones, the problem does not occur.

The problem occurs when the form is sent traditionally, jumping to a subpage after sending. I solved it by switching the form to ajax sending. Then everything is fine

**Comment by leofeyer** [comment]
Which email application do you use?

**Comment by piotrwarsztocki** [comment]
Gmail app version 2024.09.01 on Android 13

**Comment by fritzmg** [comment]
I cannot reproduce the issue. I sent an email with a link to https://demo.contao.org/applications.html to my Gmail account and opened it on my Pixel 7a. There were no issues submitting the form.

> The email application opens such links in its own frame by default

What do you mean by that? It's not a "frame", or at least not in the web sense.

**Comment by aschempp** [comment]
> What do you mean by that? It's not a "frame", or at least not in the web sense.

Probably means in a browser popup and not in the actual browser app. Not sure how Android works, but that's the case a lot on iOS.

**Comment by piotrwarsztocki** [comment]
> Probably means in a browser popup and not in the actual browser app. Not sure how Android works, but that's the case a lot on iOS.

Yes, that's what I meant. The link opens in the browser built into the mail application and not a actual dedicated browser app. The same happens when you open links in Facebook app or Messenger app. But I only had problems with Gmail app. 
I have no time to dig now. I stay with ajax sending for a while. But I will try to identify the problem once I finish the current project.

**Comment by fritzmg** [comment]
> The link opens in the browser built into the mail application and not a actual dedicated browser app.

These apps use the WebView component which will then use the default rendering engine on each system - e.g. Webkit on iOS. For all intents and purposes there is no difference between that and using the "dedicated" browser app. At least in theory.

I couldn't reproduce it on Android. Somebody else would have to try it on iOS.

**Comment by ausi** [comment]
I was not able to reproduce it on iOS 18 with GMail.

**Comment by fritzmg** [comment]
In most cases the culprit is always a separate AJAX or `fetch` request that happens after requesting the URL with the form - and said request will then cause a Cookie to be set - either because non-empty data was written to the session and thus a session cookie was set - or a custom cookie was set by the application.
